src/Raven.Server/Documents/Queries/Parse/SimpleQueryParser.cs

```diff
@@ -20,7 +15,7 @@ public static IEnumerable<string> GetTermValuesForField(IndexQueryServerSide que
             if (string.IsNullOrWhiteSpace(query.Query))
                 yield break;
 
-            var q = QueryBuilder.BuildQuery(query.Query, query.DefaultOperator, query.DefaultField, QueryAnalyzer);
+            var q = QueryBuilder.BuildQuery(query.Parsed, query.Fields.Where, QueryAnalyzer);
```

Comment:
Why do we need this now? We previously had to parse the query twice, once for the query optimizer, now we don't have to, so we can just hold to the parsed value once, no?